Marlin/Marlin_main.cpp

```diff
@@ -5510,6 +5536,19 @@ inline void gcode_M109() {
 
       float temp = thermalManager.degBed();
 
+      #if ENABLED(LED_STRIP)
+        if(wait_for_heatup) {
+          // Gradually change LED strip from yellow to red as bed heats up
+          int r, g, b;
+          float index_multiplier;
+          index_multiplier = 255 / (theTarget - 20);
+          r = 255 * (temp / index_multiplier);
+          g = r;
+          b = 255 - (temp * index_multiplier);
```

Comment:
Changing this to go from Blue to Violet then Violet to Red...makes sense to implement the map() function as it's more efficient and cleaner.